packages/dev/core/src/Engines/webgpuEngine.ts

```diff
@@ -3693,6 +3702,60 @@ export class WebGPUEngine extends ThinWebGPUEngine {
         this._currentMaterialContext.textureState = textureState;
 
         const pipeline = this._cacheRenderPipeline.getRenderPipeline(fillMode, this._currentEffect!, this.currentSampleCount, textureState);
+
+        // Compare the vertex buffers that we have to the ones that are bound to the pipeline.
+        // If there are vertex buffers that are not bound to the pipeline, AND there are storage buffers
+        // used by the shader with the same name, we will bind them to the current draw context because they
+        // are being used to do vertex pulling.
+        const availableVertexBuffers: { [key: string]: VertexBuffer } = (this._cacheRenderPipeline as any)._vertexBuffers;

```

Comment:
Would be amazing to have @Popov72  feedback on it ?

Reply:
Let me take a look (as it is, I think it's probably too much work to do in the hot path).